### PR TITLE
chore: add aws-load-balancer-provider updatecli manifest

### PR DESCRIPTION
**Pull request by lemeurherve**
Follow-up of https://github.com/jenkins-infra/aws/pull/200

### CODE REVIEW

File: updatecli/updatecli.d/aws-load-balancer-provider.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+title: Bump `aws-load-balancer-controller` version and IAM policy content
```
Comment (dduportal) [inline updatecli/updatecli.d/aws-load-balancer-provider.yaml:1]:
```suggestion
name: Bump `aws-load-balancer-controller` version and IAM policy content
```

File: updatecli/updatecli.d/aws-load-balancer-provider.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      email: "{{ .github.email }}"
+      owner: "{{ .github.owner }}"
+      repository: "{{ .github.repository }}"
+      token: "{{ requiredEnv .github.token }}"
+      username: "{{ .github.username }}"
+      branch: "{{ .github.branch }}"
+
+sources:
+  getLatestVersion:
+    kind: githubrelease
```
Comment (dduportal) [inline updatecli/updatecli.d/aws-load-balancer-provider.yaml:17]:
```suggestion
    kind: githubrelease
    name: "Retrieve the latest version"
```

File: updatecli/updatecli.d/terraform-modules/eks.yml [hunk 3]
```diff
@@ -36,14 +36,14 @@ targets:
       matchPattern: >-
         source(.*)=(.*)"terraform-aws-modules/eks/aws"(\r\n|\r|\n)(.*)version(.*)=(.*)
       replacePattern: >-
-        source${1}=${2}"terraform-aws-modules/eks/aws"${3}${4}version${5}= "{{ source `getLatestVersion` }}"
+        source${1}=${2}"terraform-aws-modules/eks/aws"${3}${4}version${5}= "{{ source "getLatestVersion" }}"
```
Comment (dduportal) [inline updatecli/updatecli.d/terraform-modules/eks.yml:39]: Can you test it please? In this context, YAML consider the string as literal, so the backticks *might* be important

File: updatecli/updatecli.d/terraform-modules/irsa.yaml [hunk 4]
```diff
@@ -36,14 +36,14 @@ targets:
...
-        source${1}=${2}"terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"${3}${4}version${5}= "{{ source `getLatestVersion` }}"
+        source${1}=${2}"terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"${3}${4}version${5}= "{{ source "getLatestVersion" }}"
     scmID: default
 
 pullrequests:
   default:
     kind: github
     scmID: default
-    title: Bump version of the Terraform module "irsa" to {{ source `getLatestVersion` }}
+    title: Bump version of the Terraform module "irsa" to {{ source "getLatestVersion" }}
```
Comment (dduportal) [inline updatecli/updatecli.d/terraform-modules/irsa.yaml:46]: Nice,  in that context, the double quotes makes sense 👍

**Comment by dduportal** [comment]
[async] Ping @lemeurherve for rebase / update

**Comment by lemeurherve** [comment]
> getLatestLockFileContent
> ERROR: environment variable "ASDF_DIR" not found, skipping

**Comment by dduportal** [comment]
> > getLatestLockFileContent
> > ERROR: environment variable "ASDF_DIR" not found, skipping

This line is prefixed as an error, true. But it is not the reason of the `updatecli diff` failure.

The failure is:

```
TARGETS
========
Pipeline "Bump `aws-load-balancer-controller` version and IAM policy content" failed
Skipping due to:
 	targets stage:	"yaml: line 39: did not find expected key"
```

https://infra.ci.jenkins.io/blue/organizations/jenkins/terraform-jobs%2Faws/detail/PR-201/5/pipeline/59#step-66-log-337

**Comment by lemeurherve** [comment]
Not sure why it still failing, any idea?

**Comment by dduportal** [comment]
Still https://infra.ci.jenkins.io/blue/organizations/jenkins/terraform-jobs%2Faws/detail/PR-201/8/pipeline#step-61-log-337

```
targets stage:	"yaml: line 39: did not find expected key"
```

**Comment by dduportal** [comment]
> > Still [infra.ci.jenkins.io/blue/organizations/jenkins/terraform-jobs%2Faws/detail/PR-201/8/pipeline#step-61-log-337](https://infra.ci.jenkins.io/blue/organizations/jenkins/terraform-jobs%2Faws/detail/PR-201/8/pipeline#step-61-log-337)
> > ```
> > targets stage:	"yaml: line 39: did not find expected key"
> > ```
> 
> Yes but why? ^^
> 
> I've tested the regexp on http://regex101.com, looks good,
> 
> > Not sure why it still failing

Played a bit locally and was able to find the cause (look at my fixup commit):

- Root cause is that you uses the `getLatestContent` source instead of the `getLatestVersion` in the replace pattern. When the templating was rendering, it injected JSON content in the YAML. Since JSON is valid in YAML, it injected content in the manifest, which the parser failed to handle (because adding keys and values).
- nitpick: fixed a warning on `dependson` and disable source input on target

**Review by dduportal — APPROVED** [review]

**Comment by timja** [comment]
I hit that error a week or two ago in updatecli, hard to debug 😢

**Comment by dduportal** [comment]
> I hit that error a week or two ago in updatecli, hard to debug 😢

Fair point, even with the `--debug` flag it was not easy to get. Had to die and retry. I guess worth an issue on updatecli to work on observability of the templating system (cc @olblak  for info)

**Comment by olblak** [comment]
> > I hit that error a week or two ago in updatecli, hard to debug cry
> 
> Fair point, even with the `--debug` flag it was not easy to get. Had to die and retry. I guess worth an issue on updatecli to work on observability of the templating system (cc @olblak for info)

It could definitely understand that it was hard to debug.
That would be great if you could open an issue. I am not sure how to fix that but I am would be curious know how it would behave with yq or dasel. The yaml resource should be refactor as it's way behind the json/csv/toml

**Comment by lemeurherve** [comment]
Issue opened.

Note: the target here is a file, not YAML

**Pull request merged**